app/BUILD.gn

```diff
@@ -0,0 +1,109 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import("//build_overrides/chip.gni")
+
+import("${chip_root}/build/chip/tools.gni")
+import("${chip_root}/src/app/common_flags.gni")
+
+assert(chip_build_tools)
+
+import("${chip_root}/examples/common/pigweed/pigweed_rpcs.gni")
+
+if (chip_enable_pw_rpc) {
+  import("//build_overrides/pigweed.gni")
+  import("$dir_pw_build/target_types.gni")
+}
+
+config("includes") {
+  include_dirs = [
+    ".",
+    "include",
+  ]
+}
+
+executable("chip-lighting-app") {
+  sources = [
+    "include/CHIPProjectAppConfig.h",
+    "include/LightingManager.h",
+    "src/LightingManager.cpp",
+    "main.cpp",
+  ]
+
+  deps = [
+    "${chip_root}/examples/lighting-app/lighting-common",
+    "${chip_root}/examples/platform/linux:app-main",
+    "${chip_root}/src/lib",
+  ]
+
+  include_dirs = [ "include" ]
+
+  if (chip_enable_pw_rpc) {
+    defines = [
+      "PW_RPC_ENABLED",
+      "PW_RPC_ATTRIBUTE_SERVICE=1",
+      "PW_RPC_BUTTON_SERVICE=1",
+      "PW_RPC_DESCRIPTOR_SERVICE=1",
+      "PW_RPC_DEVICE_SERVICE=1",
+      "PW_RPC_LIGHTING_SERVICE=1",
+      "PW_RPC_TRACING_SERVICE=1",
+    ]
+
+    sources += [
+      "${chip_root}/examples/platform/linux/Rpc.cpp",
+      "${chip_root}/examples/platform/linux/system_rpc_server.cc",
+    ]
+
+    deps += [
+      "$dir_pw_hdlc:pw_rpc",
+      "$dir_pw_hdlc:rpc_channel_output",
+      "$dir_pw_log",
+      "$dir_pw_rpc:server",
+      "$dir_pw_rpc/system_server:facade",
+      "$dir_pw_stream:socket_stream",
+      "$dir_pw_stream:sys_io_stream",
+      "$dir_pw_sync:mutex",
+      "$dir_pw_trace",
+      "$dir_pw_trace_tokenized",
+      "$dir_pw_trace_tokenized:trace_rpc_service",
+      "${chip_root}/config/linux/lib/pw_rpc:pw_rpc",
+      "${chip_root}/examples/common/pigweed:attributes_service.nanopb_rpc",
+      "${chip_root}/examples/common/pigweed:button_service.nanopb_rpc",
+      "${chip_root}/examples/common/pigweed:descriptor_service.nanopb_rpc",
+      "${chip_root}/examples/common/pigweed:device_service.nanopb_rpc",
+      "${chip_root}/examples/common/pigweed:lighting_service.nanopb_rpc",
+      "${chip_root}/examples/common/pigweed:rpc_services",
+    ]
+
+    deps += pw_build_LINK_DEPS
+
+    include_dirs += [ "${chip_root}/examples/common" ]
+  } else {
+    # The system_rpc_server.cc file is in pigweed and doesn't compile with
+    # -Wconversion, remove check for RPC build only.
+    cflags = [ "-Wconversion","-shared"]
```

Comment:
@MonicaisHer What is the `-shared` flag for? The upstream [example](https://github.com/project-chip/connectedhomeip/blob/df4ad9521b1dac98d7965f455fec74ffa270429c/examples/lighting-app/linux/BUILD.gn#L94) doesn't have it.

The linked shared objects are same with and without setting that flag:
```
ldd out/build/chip-lighting-app                                             
::      linux-vdso.so.1 (0x00007ffeafdb1000)                                         
::      libwiringPi.so => /usr/local/lib/libwiringPi.so (0x00007f5dfce86000)         
::      libcrypto.so.3 => /lib/x86_64-linux-gnu/libcrypto.so.3 (0x00007f5dfca44000)  
::      libgio-2.0.so.0 => /lib/x86_64-linux-gnu/libgio-2.0.so.0 (0x00007f5dfc86c000)
::      libgobject-2.0.so.0 => /lib/x86_64-linux-gnu/libgobject-2.0.so.0 (0x00007f5dfc80c000)                                                                          
::      libglib-2.0.so.0 => /lib/x86_64-linux-gnu/libglib-2.0.so.0 (0x00007f5dfc6d2000)                                                                                
::      libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f5dfc4a4000)  
::      libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f5dfc3bd000)            
::      libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f5dfc195000)            
::      libcrypt.so.1 => /lib/x86_64-linux-gnu/libcrypt.so.1 (0x00007f5dfc15b000)    
::      libgmodule-2.0.so.0 => /lib/x86_64-linux-gnu/libgmodule-2.0.so.0 (0x00007f5dfc154000)                                                                          
::      libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f5dfc138000)            
::      libmount.so.1 => /lib/x86_64-linux-gnu/libmount.so.1 (0x00007f5dfc0f2000)    
::      libselinux.so.1 => /lib/x86_64-linux-gnu/libselinux.so.1 (0x00007f5dfc0c6000)
::      libffi.so.8 => /lib/x86_64-linux-gnu/libffi.so.8 (0x00007f5dfc0b9000)        
::      libpcre.so.3 => /lib/x86_64-linux-gnu/libpcre.so.3 (0x00007f5dfc043000)      
::      /lib64/ld-linux-x86-64.so.2 (0x00007f5dfd12e000)                             
::      libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f5dfc023000)    
::      libblkid.so.1 => /lib/x86_64-linux-gnu/libblkid.so.1 (0x00007f5dfbfea000)    
::      libpcre2-8.so.0 => /lib/x86_64-linux-gnu/libpcre2-8.so.0 (0x00007f5dfbf53000)
```

Reply:
It was added for creating a shared wiringPi library: https://github.com/MonicaisHer/connectedhomeip/commit/e22c0e43fe45a735b40b9ca9134c991c18b87302

Reply:
Okay. I removed it as it has no obvious effect.